system-test/storage.ts

```diff
@@ -2250,6 +2250,37 @@ describe('storage', () => {
       });
     });
 
+    it('should skip validation if file is served decompressed', done => {
+      const filename = 'logo-gzipped.png';
+      bucket
+        .upload(FILES.logo.path, {
+          destination: filename,
+          gzip: true,
+        })
+        .then(() => {
+          tmp.setGracefulCleanup();
+          tmp.file((err, tmpFilePath) => {
+            assert.ifError(err);
+
+            const file = bucket.file(filename);
+            file
+              .createReadStream()
```

Comment:
This needs an on-error handler.

Reply:
Different from the one on line 2269?

Reply:
Yeah that one will catch errors from the write stream, but if the read stream has an error, it would go uncaught currently.

Reply:
Looks like @jkwlui addressed this?